pyrtl/rtllib/multipliers.py

```diff
@@ -172,18 +171,20 @@ def signed_tree_multiplier(A, B, reducer=adders.wallace_reducer, adder_func=adde
             bits[i + j].append(x & y)
 
     res = reducer(bits, bits_length+1, adder_func)
```

Comment:
It's not the same code, `tree_mult` uses bit_length`, not `bit_length+1`. Also it uses `_trivial_mult()` when one of the inputs is 1 bit, while the signed version is supposed to throw an error there.